### PR TITLE
Add checkstyle

### PR DESCRIPTION
**Pull request by kinbiko**
Addresses #21 

Adds some style linting to Budjen. This means that if we violate any of the rules in this ruleset (defined by the configuration in `checkstyle.xml`) we will fail the build.

I've based this off of Google's checkstyle configuration, and removed the checks I disagree with (which would have made this PR a lot bigger!). I've also addressed the checks I agree with that we were violating - hence the changes to some of the source code.

### CODE REVIEW

File: src/main/java/com/missafternoon/budjen/transactions/Validator.java [hunk 1]
```diff
@@ -10,16 +10,16 @@ public boolean validate(String[] args) {
                 && contains(args, "--description")
                 && contains(args, "--amount")
                 && (contains(args, "--credit") ^ contains(args, "--debit"))
-        )) {
+            )) {
```
Comment (JStonham) [inline src/main/java/com/missafternoon/budjen/transactions/Validator.java]: Possibly the indentation discrepancy? To be honest, this is just a guess!

Comment (kinbiko) [reply src/main/java/com/missafternoon/budjen/transactions/Validator.java]: Nope, sorry. This was fixing a checkstyle violation: The indentation of `)` didn't match the indentation of `(`.

File: build.gradle [hunk 2]
```diff
@@ -23,6 +24,10 @@ repositories {
 	mavenCentral()
 }
 
+checkstyle {
+	toolVersion = "8.16"
+  configFile = new File(rootDir, "checkstyle.xml")
```
Comment (kinbiko) [inline build.gradle]: @JStonham This is the one! Can you tell what the difference is between this line and the one above? What if you open it in IntelliJ, can you see a difference then?

Comment (JStonham) [reply build.gradle]: Oh!! I confused myself! I was thinking it had to be related to brackets, but you didn't suggest that - I did!

**Pull request merged**